src/injection.ts

```diff
@@ -25,10 +32,11 @@ export function autoinject(
       }
     }
   };
-
-  return potentialTarget ? deco(potentialTarget) : deco;
+  if (isInjectable(potentialTarget)) {
+    return deco(potentialTarget) as TPotential extends Injectable ? void : (target: Injectable) => void;
```

Comment:
This cast shouldn't be necessary right?

```suggestion
    return deco(potentialTarget);
```